src/SelfServe/Documentation/README.md

```diff
@@ -0,0 +1,197 @@
+# Self Serve Model
+
+The Self Serve Model allows you to write classes that auto generate UI components for your feature. The idea is to allow developers, who aren't very familiar with writing UI, to develop and own UI components for their features by just writing simpler TypeScript classes. 
+
+What this means for the feature team 
+- Can concentrate just on the logic behind showing, hiding and disabling UI components 
+- Need not worry about specifics of the UI language
+- Can own the REST API calls made as part of the feature, which may change in the future
+- Quicker turn around time for feature bugs, which they own and have deeper knowledge of
+
+What this means for the UI team
+- Lesser feature development time, since they only own the framework and not the feature itself
+
+Each team owns what they know best, a win-win!
+
+## Getting Started
+
+Clone the cosmos-explorer repo and run
+
+- `npm install`
+- `npm run build`
+
+## Developing
+
+### Watch mode
+
+Run `npm start` to start the development server and automatically rebuild on changes
+
+### Local Development
+
+Update [SelfServeType](../enums/selfserve_selfserveutils.selfservetype.html) to add your feature.
+
+Add the feature flag `feature.dataExplorerSource=https://localhost:1234/selfServe.html?selfServeType%3D<SELF_SERVE_TYPE>` to open up the your feature's UI in the data explorer blade of the portal.
+
+For example, if you want to open up the SelfServeExample UI, visit `https://ms.portal.azure.com/?feature.dataExplorerSource=https://localhost:1234/selfServe.html?selfServeType%3Dexample`
+
+![](https://sdkctlstore.blob.core.windows.net/exe/selfserveDevelopment.PNG)
+
+
+## Code Changes
```

Comment:
>Changes [](http://example.com/codeflow?start=8&length=7)

Add text saying: we need changes in only 4 files with simple iverview fo what kind of chnages go in that 4 files.